p4-16/psa/Makefile

```diff
@@ -40,23 +41,10 @@ check:
 # psa-example-mirror-on-drop.p4 needs updates for latest psa.p4
 #	${P4C} examples/psa-example-mirror-on-drop.p4
 
-# psa-example-incremental-checksum2.p4 fails with latest p4test as of
-# 2017-Dec-04 because of call to InternetChecksum method set_state(),
-# because its argument is not a compile-time constant.  I am not sure
-# why that should be a problem.
-#	${P4C} examples/psa-example-incremental-checksum2.p4
-
 # psa-example-register1.p4 is almost identical to
 # psa-example-register2.p4, except that it attempts to return a struct
 # from a register read, and write a struct back to the register.  As
-# of 2017-Dec-04, the latest version of p4test gives an error if you
-# attempt this.
+# of 2018-Oct-19, the latest version of p4test gives an error if you
+# attempt this, I believe somewhere in one of the mid end passes.  It
+# is definitely not in the bmv2 back end code.
```

Comment:
We should track this down and fix. I'll take a look before the release.

Reply:
I don't think bmv2 supports registers with structs.
We can implement this in the compiler by using bit slices and concatenations if it is necessary, but I am not sure how useful this will be in general. The main question is about field alignment in a register and what hardware can support in general.

Reply:
I think it would be nice to have, but realize there are other useful things to work on besides this.

The example program was the first one I tried to write using registers with multiple 'fields' inside of the register array elements, and when it did not work I left the example around as a reminder, but did not include it in the PSA spec document, including only the variant of the program that used explicit bit slices and concatentation instead, since that compiles with p4c.  I can create a separate issue for p4c for it, with Mihai's comment, in case that helps to track it.  It sounds like it may require enhancements in bmv2 as well.

Reply:
Created this issue just to record it somewhere closer to where changes would be needed to make the enhancement: https://github.com/p4lang/p4c/issues/1591